diff --git a/source3/utils/net_ads.c b/source3/utils/net_ads.c
index 3cf8fbbf7c8a..9d1b8669c021 100644
--- a/source3/utils/net_ads.c
+++ b/source3/utils/net_ads.c
@@ -3551,7 +3551,7 @@ static int net_ads_enctype_lookup_account(struct net_context *c,
 	int ret = -1;
 	ADS_STATUS status;
 
-	filter = talloc_asprintf(c, "(&(objectclass=user)(sAMAccountName=%s))",
+	filter = talloc_asprintf(c, "(&(|(objectclass=computer)(objectclass=user))(sAMAccountName=%s))",
 				 account);
 	if (filter == NULL) {
 		goto done;
