diff --git a/nsswitch/wscript_build b/nsswitch/wscript_build
index 4e62bb4c9461..e77373a8cc9a 100644
--- a/nsswitch/wscript_build
+++ b/nsswitch/wscript_build
@@ -41,7 +41,7 @@ if (Utils.unversioned_sys_platform() == 'linux' or (host_os.rfind('gnu') > -1)):
               source='winbind_nss_linux.c',
               deps='wbclient',
               realname='libnss_winbind.so.2',
-              soname='libnss_winbind.so.2',
+              soname='libnss_winbind.so',
               vnum='2')
 
     bld.SAMBA3_PLUGIN('nss_wins',
@@ -50,7 +50,7 @@ if (Utils.unversioned_sys_platform() == 'linux' or (host_os.rfind('gnu') > -1)):
                       source='wins.c',
                       deps='wbclient replace',
                       realname='libnss_wins.so.2',
-                      soname='libnss_wins.so.2',
+                      soname='libnss_wins.so',
                       vnum='2')
 elif (host_os.rfind('freebsd') > -1):
 	# FreeBSD winbind client is implemented as a wrapper around
